FWCore/Framework/src/EventProcessor.cc

```diff
@@ -1744,22 +1744,18 @@ namespace edm {
                           looper_->prefetchAsync(
                               nextTask, serviceToken_, Transition::BeginLuminosityBlock, *(status->lumiPrincipal()), es);
                         }) | ifThen(looper_, [this, status, &es](auto nextTask) {
-                          status->globalBeginDidSucceed();
                           ServiceRegistry::Operate operateLooper(serviceToken_);
                           looper_->doBeginLuminosityBlock(*(status->lumiPrincipal()), es, &processContext_);
                         }) | then([this, status, iRunStatus](std::exception_ptr const* iException, auto holder) mutable {
+                          status->setGlobalEndRunHolder(iRunStatus->globalEndRunHolder());
+
                           if (iException) {
-                            status->resetResources();
-                            queueWhichWaitsForIOVsToFinish_.resume();
                             WaitingTaskHolder copyHolder(holder);
                             copyHolder.doneWaiting(*iException);
+                            globalEndLumiAsync(holder, status);
```

Comment:
Do I understand correctly the `globalEndLumiAsync()` calls the `status->resetResources()` and `queueWhichWaitsForIOVsToFinish_.resume()`? Does the `globalEndLumiAsync()` do anything else that is good to be called at this point is not done presently?

Reply:
Presently, we do not run the global end luminosity block transition if the global begin luminosity block transition fails. For example, if any module throws in global begin luminosity block then we run none of the modules at global end lumi and also none of the signals or anything else.

Adding this line that calls globalEndLumiAsync causes the global end luminosity block transition to occur even when there is an exception thrown in global begin luminosity block. That is why I added this line. Not needing to call ```resetResources``` and ```queueWhichWaitsForIOVsToFinish_.resume()``` was just a side effect of that. 

Independently, in Worker.h there is new code to inhibit the individual module that threw in global begin lumi to not run in global end lumi.